### PR TITLE
Improve REPL completion for class names and constructors

### PR DESCRIPTION
**Pull request by retronym**
```
Welcome to Scala 2.12.9-20190308-043616-e9bd65d (OpenJDK 64-Bit Server VM, Java 1.8.0-adoptopenjdk).
Type in expressions for evaluation. Or try :help.

scala> class Shazam(i :Int) { def this(s: String)(i: Int) = this(0) }
defined class Shazam

scala> new Sh
Shazam   Short

scala> new Shazam
   def <init>(i: Int): Shazam   def <init>(s: String)(i: Int): Shazam

scala> new Shazam
```

### CODE REVIEW

File: src/repl/scala/tools/nsc/interpreter/PresentationCompilation.scala [hunk 1]
```diff
@@ -99,9 +99,13 @@ trait PresentationCompilation {
 
     def completionsAt(cursor: Int): CompletionResult = {
       val pos = unit.source.position(preambleLength + cursor)
```
Comment (dwijnand) [inline src/repl/scala/tools/nsc/interpreter/PresentationCompilation.scala]: unused

**Review by dwijnand — COMMENTED** [review]
:+1:

**Comment by retronym** [comment]
The next feature we need to support for showing the "def string" after an applied type tree.

`new Some[String]<TAB>` or `identity[String]<TAB>`.

File: src/compiler/scala/tools/nsc/typechecker/Contexts.scala [hunk 2]
```diff
@@ -1567,7 +1567,7 @@ trait Contexts { self: Analyzer =>
     private def transformImport(selectors: List[ImportSelector], sym: Symbol): List[Symbol] = selectors match {
       case List() => List()
       case List(ImportSelector(nme.WILDCARD, _, _, _)) => List(sym)
-      case ImportSelector(from, _, to, _) :: _ if from == sym.name =>
+      case ImportSelector(from, _, to, _) :: _ if from == (if (from.isTermName) sym.name.toTermName else sym.name.toTypeName) =>
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/typechecker/Contexts.scala:1570]: Does `Name` deserve `name_==` alongside weird `string_==`?

Comment (retronym) [reply src/compiler/scala/tools/nsc/typechecker/Contexts.scala:1570]: I can't think of cases when it makes sense to equate Term/Type-names outside of import selectors, so I opted not to introduce a method on `Name` that might be misused.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/typechecker/Contexts.scala:1570]: OK, next time I'll just use `string_==` 😺

**Pull request merged**